packages/forms-web-app/src/dynamic-forms/controller.js

```diff
@@ -238,10 +238,23 @@ exports.submit = async (req, res) => {
 	const { referenceId } = req.params;
 	const journeyResponse = res.locals.journeyResponse;
 	const journey = getJourney(journeyResponse);
+
+	const journeyUrl = (journeyId) => {
```

Comment:
In situations like this (where you've got a 1:1 mapping from one string to another and you're likely to add more things to it later) it's often preferable to use a dictionary style approach rather than having many bits of branching logic. If you made an object like

```
const journeyIdToUrl = {
  'has-questionnaire': 'householder/',
  's78-questionnaire': 'full-planning'
}
```

then, so long as you verify that the given id is valid, you can unconditionally return the matching element from the dictionary 

```
'/manage-appeals/' +
				journeyIdToUrl[journeyResponse.journeyId] +
				encodeURIComponent(referenceId) +
				'/questionnaire-submitted/'
```

Reply:
Just going to get this through for now, but thank you, this is a great point! I'll look into fixing it at a later date